Formula/llvm.rb

```diff
@@ -188,17 +226,36 @@ def install
     args = %w[
       -DLLVM_OPTIMIZED_TABLEGEN=On
       -DLLVM_BUILD_LLVM_DYLIB=On
+      -DLLVM_TARGETS_TO_BUILD=X86
     ]
 
+    args << "-DLLVM_BUILD_EXTERNAL_COMPILER_RT=On" if build.with? "compiler-rt"
+    if build.with? "test"
+      args << "-DLLVM_BUILD_TESTS=On"
+      args << "-DLLVM_ABI_BREAKING_CHECKS=On"
+    end
     args << "-DLLVM_ENABLE_RTTI=On" if build.with? "rtti"
     args << "-DLLVM_INSTALL_UTILS=On" if build.with? "utils"
+    args << "-DLLVM_ENABLE_LIBCXX=On" if build.with? "libcxx" || (build.with? "clang" && !MacOS::CLT.installed?)
+    args << "-DLLVM_ENABLE_LIBCXXABI=On" if build.with? "libcxxabi"
+    args << "-DLLVM_ENABLE_DOXYGEN=On" if build.with? "doxygen"
+    args << "-DBUILD_SHARED_LIBS=On" if build.with? "shared-libs" # for developers
+
+    if build.with? "libffi"
+      args << "-DLLVM_ENABLE_FFI=On"
+      args << "-DFFI_INCLUDE_DIR=#{Formula["libffi"].lib}/libffi-#{Formula["libffi"].version}/include" # upstream bug?
```

Comment:
Well, it means that naïve `-DFFI_INCLUDE_DIR=#{Formula["libffi"].include}` doesn't work. `Libffi`'s include directory is burried in its library directory (see the path).